### PR TITLE
[ticket/13700] Add method for initializing config after cache purge

### PR DESCRIPTION
**Pull request by marc1706**
PHPBB3-13700

The goal of this is to be able to re-initialize the config entries after a cache purge, e.g. when running db migrations via CLI.

Checklist:

- [x] Correct branch: master for new features; 3.3.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/dev/master/development/coding_guidelines.html) and [3.3.x](https://area51.phpbb.com/docs/dev/3.3.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.3.x/development/git.html)

Tracker ticket (set the ticket ID to **your ticket ID**):

https://tracker.phpbb.com/browse/PHPBB3-13700

### CODE REVIEW

File: phpBB/phpbb/config/db.php [hunk 1]
```diff
@@ -39,16 +42,28 @@ class db extends \phpbb\config\config
...
+	/**
+	 * Initialise config with database and/or cached entries
+	 *
+	 * @param cache_interface $cache
+	 */
+	public function initialise(cache_interface $cache)
+	{
 		if (($config = $cache->get('config')) !== false)
 		{
 			$sql = 'SELECT config_name, config_value
```
Comment (rxu) [inline phpBB/phpbb/config/db.php:69]: During installation process, `phpbb_config` table does not exist until some point, so this SQL query causes hidden (not reported) SQL error.
Probably needs a check for `_config` table existence and initializing `$config` as an empty array if the table doesn't exist.

Comment (marc1706) [reply phpBB/phpbb/config/db.php:69]: I take it this was already an issue before this PR then?

Comment (rxu) [reply phpBB/phpbb/config/db.php:69]: Yeah, but looks like a good opportunity to fix it while changing the same code snippet :)

**Comment by 3D-I** [comment]
Is PHPBB3-13700 the right ticket?

**Comment by marc1706** [comment]
I think so based on https://tracker.phpbb.com/browse/PHPBB3-13700?focusedCommentId=59504&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-59504 and the fact that I was able to reproduce issues in a similar fashion that resulted in the problems described in that ticket.

File: phpBB/phpbb/config/db.php (hunk 1 above)
Comment (marc1706) [reply phpBB/phpbb/config/db.php:69]: Created a separate ticket for this: https://tracker.phpbb.com/browse/PHPBB3-16806

**Pull request merged**